File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let l=0;
    for(let i in nums){
        if(nums[i]!=0){
            let temp=nums[l]
            nums[l]=nums[i]
            nums[i]=temp
            l++;
        }
        // if(nums[l]!=0){
        //     l++
        // }
    }
    
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int l=0;
        for(int i=0;i<nums.length;i++){
            if(nums[l]==0 && nums[i]!=0){
                int temp=nums[l];
                nums[l]=nums[i];
                nums[i]=temp;
                l++;
            }
            if(nums[l]!=0){
                l++;
            }
        }
    }
}